src/components/Form.js

```diff
@@ -1,61 +1,16 @@
-import TaskList from "../tasks/TaskList";
-import NoTaskImage from "./NoTaskImage";
-import Toast from "./Toast";
-
-import { useState, useEffect } from "react";
+import DataContext from "../store/DataContext";
 
+import { useEffect, useContext } from "react";
 
 function Form() {
-    const [input, setInput] = useState("");
-    const [tasks, setTasks] = useState([]);
-    const [toastStatus, setToastStatus] = useState(false);
-
-    //obtaining the value from the input field
-    const handleInput = (e) => setInput(e.target.value);
-
-    //adding the values to the tasks array
-    const handleSetTasks = () => {
-        const newTask = {
-            text: input,
-            id: Date.now(),
-        };
-
-        if (input.length) {
-            setTasks([...tasks, newTask]);
-        }
-        setInput("");
-    };
-
-    //Displaying Toast
-    const handleSetToastStatus = () => {
-        setToastStatus(true);
-
-        setTimeout(() => {
-            setToastStatus(false);
-        }, 800);
-    };
-
-    //logging the array for reference
-    useEffect(() => {
-        if (tasks.length) {
-            console.log("changing");
-            console.log(tasks);
-        }
-    }, [tasks]);
-
-    //deleting completed tasks on clicking
-    const deleteCompletedTask = (clickedObj) => {
-        const updatedArray = tasks.filter((obj) => obj.id !== clickedObj.id);
-        setTasks(updatedArray);
-        handleSetToastStatus();
-    };
+    const data = useContext(DataContext); //using context
 
     //listening for enter-key press and calling the handleSetTasks function
     useEffect(() => {
```

Comment:
no need for this event listener. We can use the onSubmit function of the  form.

Reply:
used onSubmit